homeassistant/components/coinbase/__init__.py

```diff
@@ -1,38 +1,43 @@
-"""Support for Coinbase."""
+"""The Coinbase integration."""
+import asyncio
 from datetime import timedelta
 import logging
 
 from coinbase.wallet.client import Client
 from coinbase.wallet.error import AuthenticationError
 import voluptuous as vol
 
-from homeassistant.const import CONF_API_KEY
+from homeassistant.config_entries import SOURCE_IMPORT, ConfigEntry
+from homeassistant.const import CONF_API_KEY, CONF_API_TOKEN
+from homeassistant.core import HomeAssistant
 import homeassistant.helpers.config_validation as cv
-from homeassistant.helpers.discovery import load_platform
 from homeassistant.util import Throttle
 
-_LOGGER = logging.getLogger(__name__)
-
-DOMAIN = "coinbase"
+from .const import (
+    API_ACCOUNT_ID,
+    CONF_YAML_API_KEY,
+    CONF_YAML_API_TOKEN,
+    CONF_YAML_CURRENCIES,
+    CONF_YAML_EXCHANGE_RATES,
+    DOMAIN,
+)
 
-CONF_API_SECRET = "api_secret"
-CONF_ACCOUNT_CURRENCIES = "account_balance_currencies"
-CONF_EXCHANGE_CURRENCIES = "exchange_rate_currencies"
+_LOGGER = logging.getLogger(__name__)
 
+PLATFORMS = ["sensor"]
 MIN_TIME_BETWEEN_UPDATES = timedelta(minutes=1)
 
-DATA_COINBASE = "coinbase_cache"
 
 CONFIG_SCHEMA = vol.Schema(
     {
         DOMAIN: vol.Schema(
```

Comment:
As we are now importing and migrating YAML configuration, a deprecation warning should be thrown for it. Please mark the schema deprecated.